<ui version="4.0" >
 <class>Config</class>
 <widget class="QWidget" name="Config" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>532</width>
    <height>441</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout" >
   <item>
    <widget class="QGroupBox" name="General" >
     <property name="title" >
      <string>General options</string>
     </property>
     <layout class="QVBoxLayout" >
      <item>
       <widget class="QCheckBox" name="UseWeight" >
        <property name="toolTip" >
         <string>specifies if the meta engine must use weight defined for each engine( see engine configuration) or use an identical weight for every engine</string>
        </property>
        <property name="text" >
         <string>Use Engine weight</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="_2" >
        <item>
         <widget class="QCheckBox" name="LimitRes" >
          <property name="text" >
           <string>Limits the number of results to be displayed : </string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="spacer5_3" >
          <property name="orientation" >
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType" >
           <enum>QSizePolicy::Expanding</enum>
          </property>
          <property name="sizeHint" stdset="0" >
           <size>
            <width>61</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="KIntSpinBox" name="MaxResNb" >
          <property name="toolTip" >
           <string>if the limitation is used, no more than "X" results will be displayed</string>
          </property>
          <property name="maxValue" stdset="0" >
           <number>1000</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="_3" >
        <item>
         <widget class="QLabel" name="textLabel1" >
          <property name="text" >
           <string>The number of keywords to use to create a query :</string>
          </property>
          <property name="wordWrap" >
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="spacer6" >
          <property name="orientation" >
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType" >
           <enum>QSizePolicy::Expanding</enum>
          </property>
          <property name="sizeHint" stdset="0" >
           <size>
            <width>10</width>
            <height>21</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="KIntSpinBox" name="NValue" />
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="QueryModeOption" >
     <property name="toolTip" >
      <string>Choose the working mode of the meta engine</string>
     </property>
     <property name="whatsThis" >
      <string>The General Option allows the user to choose between different working mode for the &lt;b>Meta Engine&lt;/b> :&lt;br>
&lt;u>&lt;b>1.Single Query Mode:&lt;/b>&lt;/u>
&lt;p> The set of keywords received by the meta engine is used in is totality to create one query.&lt;/p>
&lt;u>&lt;b>2. Combinaison of query terms:&lt;/b>&lt;/u>
&lt;p>The set of keywords received by the meta engine is used to create several combinaison of the keywords. All these combinaisons are used as query. The meta engine retreive the results for each combinaison of keywords and the rank all the results together. &lt;br>
The &lt;b>among&lt;/b> of combinaison is determiner by the PUpper Value witch determine the "p" value for a combinaison Cnp where "n" is the number of keywords retreived by the meta engine&lt;/p>
&lt;u>&lt;b>3. Ranch of combinaisons of the query terms&lt;/b>&lt;/u>
&lt;p>This option enable the meta engine to use a ranch of combinaisons (defined as in point 2). Two Value must be specified for the "p" value, the lower value and the upper value. Combinaisons are then create using as value "p" all the value between the two threshold&lt;/p></string>
     </property>
     <property name="title" >
      <string>Query Mode option</string>
     </property>
     <property name="checkable" >
      <bool>false</bool>
     </property>
     <property name="checked" >
      <bool>false</bool>
     </property>
     <layout class="QGridLayout" name="_4" >
      <item row="2" column="0" >
       <widget class="QCheckBox" name="checkBox3" >
        <property name="toolTip" >
         <string>the queries are formed using different combinaison of the keywords (P is the parameter for the combinaison)</string>
        </property>
        <property name="text" >
         <string>Use a Ranch of combinaisons of the query terms</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" >
       <widget class="QCheckBox" name="SingleQuery" >
        <property name="toolTip" >
         <string>All keywords are used to create one query</string>
        </property>
        <property name="text" >
         <string>Use Single query</string>
        </property>
        <property name="checked" >
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0" >
       <widget class="QCheckBox" name="CombineQuery" >
        <property name="toolTip" >
         <string>the queries are formed using a combinaison of the keywords (P is the parameter for the combinaison)</string>
        </property>
        <property name="text" >
         <string>Use a Combinaison of the query terms</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="CombinaisonOptions" >
     <property name="whatsThis" >
      <string>This panel able the configuration of the parameters used to generate combinaison of the query.&lt;br>&lt;br>
The "&lt;b>P Lower &lt;/b>" Value represents the number of keywords used to generate one query. The mete engine creates all the possible combinaison to put "P" keywords from the given set into the formed query. All the queries are processed and the results are ranked.&lt;br>
The "&lt;b>P Upper&lt;/b>" Value is used when ranch of combinaison must be retreived.The meta engine create all the queries for all the diferents combinaisons with "P" value contained between "P Lower" and "P Upper".&lt;br>&lt;br>
Values of "P" can be defined in two ways:&lt;br>
&lt;u>&lt;b>1. Percentage value:&lt;/b>&lt;/u>
The values of "P" are expressed as a percentage of the number of keywords available in the keywords set received by the meta engine.&lt;br>
&lt;u>&lt;b>2. Absolute value:&lt;/b>&lt;/u>
The values of "P" are expressed as absolute values. If the values of P are greater than the number of keywods received by the engine, the values of "P" will be remplaced by this value.</string>
     </property>
     <property name="title" >
      <string>Combinaison options</string>
     </property>
     <layout class="QGridLayout" name="_5" >
      <item row="2" column="0" >
       <widget class="QCheckBox" name="UseAbsValue" >
        <property name="text" >
         <string>Use Absolute value</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" >
       <layout class="QHBoxLayout" name="_6" >
        <item>
         <widget class="QLabel" name="PLowerPCLabel" >
          <property name="text" >
           <string>Lower value fo P :</string>
          </property>
          <property name="wordWrap" >
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="KIntSpinBox" name="PLowerPC" >
          <property name="maxValue" stdset="0" >
           <number>100</number>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="spacer5" >
          <property name="orientation" >
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType" >
           <enum>QSizePolicy::Expanding</enum>
          </property>
          <property name="sizeHint" stdset="0" >
           <size>
            <width>68</width>
            <height>21</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="PUpperPCLabel" >
          <property name="text" >
           <string>Upper value of P :</string>
          </property>
          <property name="wordWrap" >
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="KIntSpinBox" name="PUpperPC" >
          <property name="maxValue" stdset="0" >
           <number>100</number>
          </property>
          <property name="lineStep" stdset="0" >
           <number>1</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="3" column="0" >
       <layout class="QHBoxLayout" name="_7" >
        <item>
         <widget class="QLabel" name="PLowerAbsLabel" >
          <property name="text" >
           <string>Lower value fo P :</string>
          </property>
          <property name="wordWrap" >
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="KIntSpinBox" name="PLowerAbs" />
        </item>
        <item>
         <spacer name="spacer5_2" >
          <property name="orientation" >
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType" >
           <enum>QSizePolicy::Expanding</enum>
          </property>
          <property name="sizeHint" stdset="0" >
           <size>
            <width>68</width>
            <height>21</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="PUpperAbsLabel" >
          <property name="text" >
           <string>Upper value of P :</string>
          </property>
          <property name="wordWrap" >
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="KIntSpinBox" name="PUpperAbs" />
        </item>
       </layout>
      </item>
      <item row="0" column="0" >
       <widget class="QCheckBox" name="UsePercentValue" >
        <property name="text" >
         <string>Use percentage value</string>
        </property>
        <property name="checked" >
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KIntSpinBox</class>
   <extends>QSpinBox</extends>
   <header>knuminput.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
